assets/js/components/GoogleChart.js

```diff
@@ -19,205 +19,186 @@
 /**
  * External dependencies
  */
-import debounce from 'lodash/debounce';
-import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Chart } from 'react-google-charts';
 
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
-import { useEffect, useState, useRef } from '@wordpress/element';
+import { useEffect, useLayoutEffect, useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import ProgressBar from './ProgressBar';
+import PreviewBlock from './PreviewBlock';
 
 export default function GoogleChart( props ) {
 	const {
-		chartID,
+		chartEvents,
 		chartType,
+		children,
 		className,
 		data,
-		loadCompressed,
-		loadHeight,
-		loadSmall,
-		loadText,
+		getChartWrapper,
+		height,
+		loaded,
+		loadingHeight,
+		loadingWidth,
+		onMouseOver,
+		onMouseOut,
 		onReady,
-		options,
+		onSelect,
 		selectedStats,
-		singleStat,
+		width,
+		...otherProps
 	} = props;
 
-	const chartRef = useRef( null );
-	const [ chart, setChart ] = useState( null );
-	const [ loading, setLoading ] = useState( true );
-	const [ visualizationLoaded, setVisualizationLoaded ] = useState( false );
-
-	// Create a new chart when the library is loaded.
-	useEffect( () => {
-		if ( ! chart && ! loading && chartRef.current && visualizationLoaded ) {
-			let googleChart;
-
-			if ( ! chartID || ! GoogleChart.charts.has( chartID ) ) {
-				switch ( chartType ) {
-					case 'area':
-						googleChart = new global.google.visualization.AreaChart( chartRef.current );
-						break;
-					case 'line':
-						googleChart = new global.google.visualization.LineChart( chartRef.current );
-						break;
-					case 'pie':
-						googleChart = new global.google.visualization.PieChart( chartRef.current );
-						break;
-					default:
-						throw new Error( 'Unknown chart type' );
-				}
-
-				const chartData = { chart: googleChart };
-				if ( onReady ) {
-					chartData.onReady = global.google.visualization.events.addListener( googleChart, 'ready', onReady );
-				}
-
-				if ( chartID ) {
-					GoogleChart.charts.set( chartID, chartData );
-				}
-			} else {
-				googleChart = GoogleChart.charts.get( chartID ).chart;
-			}
+	// Ensure we don't filter out columns that aren't data, but are things like
+	// tooltips or other content.
+	const nonDataColumns = data?.length ? data[ 0 ].reduce( ( acc, row, rowIndex ) => {
+		return row?.role ? [ ...acc, rowIndex ] : acc;
+	}, [] ) : [];
```

Comment:
Too many question marks and colons here. Could you, please, re-format it to make it possible to ready? :smile: Maybe something like this will do it better:

```suggestion
	const nonDataColumns = data?.length 
		? data[ 0 ].reduce( ( acc, row, rowIndex ) => row?.role ? [ ...acc, rowIndex ] : acc }, [] )
		: [];
```